src/fitnesse/wikitext/parser/decorator/SlimTableDefaultColoring.java

```diff
@@ -0,0 +1,142 @@
+package fitnesse.wikitext.parser.decorator;
+
+import fit.FixtureLoader;
+import fit.FixtureName;
+import fitnesse.testsystems.slim.tables.DecisionTable;
+import fitnesse.testsystems.slim.tables.DynamicDecisionTable;
+import fitnesse.testsystems.slim.tables.ImportTable;
+import fitnesse.testsystems.slim.tables.QueryTable;
+import fitnesse.testsystems.slim.tables.ScenarioTable;
+import fitnesse.testsystems.slim.tables.ScriptTable;
+import fitnesse.testsystems.slim.tables.SlimTable;
+import fitnesse.testsystems.slim.tables.SlimTableFactory;
+import fitnesse.util.ClassUtils;
+import fitnesse.wikitext.parser.Maybe;
+import fitnesse.wikitext.parser.Symbol;
+import fitnesse.wikitext.parser.Table;
+import fitnesse.wikitext.parser.VariableSource;
+
+import java.util.List;
+
+import static fitnesse.wikitext.parser.decorator.SymbolClassPropertyAppender.classPropertyAppender;
+import static fitnesse.wikitext.parser.decorator.SymbolInspector.inspect;
+
+public class SlimTableDefaultColoring implements ParsedSymbolDecorator {
+
+  private static final String DISABLE_DEFAULT_COLORING_MARKER_PROP = "__disable_default_coloring_";
+
+  private static boolean isInstalled;
+
+  public static void install() {
+    if (!isInstalled) {
+      Table.symbolType.prependDecorator(new SlimTableDefaultColoring());
+      isInstalled = true;
+    }
+  }
+
+  //visible for testing
+  SlimTableDefaultColoring() {
+    //hidden
+  }
+
+  /**
+   * Can be called by preceding decorators to override the coloring logic for a specific table.
+   */
+  public static void disableForTable(Symbol table) {
```

Comment:
Would it not make more sense to have a feature to remove this decorator from the decorators list for this table, instead of always applying it but not doing anything if this method is called?

Reply:
The decorators list is on the _symbol type_ not _symbol_. This method allows disabling the decoration for a particular _symbol_ by adding a marker property to the _symbol_. Obviously this only works when the decoration to be disabled comes **after** the decoration that is disabling it (therefore the `prependDecorator()` method).

For well written decorators added by plugins, disabling should not be required (you can always remove the plugin if you wish).
The `SlimTableDefaultColoring` decorator is special in that it's always enabled and you cannot disable it by uninstalling a plugin.

As an alternative solution, I can imagine that the user would define a special environment property that disables the default decorator completely - it would never be added to the decorator list.